### ISSUE TITLE
Untyped code inference regression

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1360]
**Describe the bug**
With `useLibraryCodeForTypes` enabled, Pyright versions prior to 1.1.100 were able to infer the return type of untyped functions in libraries on a fairly consistent basis.  With 1.1.100, I've observed that this is no longer the case with a couple of libraries that I depend on.

**To Reproduce**
Install `jellyfish`, set `useLibraryCodeForTypes` in pyrightconfig.json to `true` and type check the snippet below in 1.1.99 and 1.1.100:

```
$ npx pyright test.py
npx: installed 1 in 3.308s
Loading configuration file at /[...]/test/pyrightconfig.json
Assuming Python version 3.9
Assuming Python platform Darwin
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding .git
stubPath /[...]/test/typings is not a valid directory.
Searching for source files
Found 1 source file
/[...]/test/test.py
  3:13 - info: Type of "jellyfish.levenshtein_distance" is "(s1: Unknown, s2: Unknown) -> Unknown"
  4:13 - info: Type of "jellyfish.jaro_similarity" is "(s1: Unknown, s2: Unknown) -> Unknown"
  5:13 - info: Type of "jellyfish.jaro_winkler_similarity" is "(s1: Unknown, s2: Unknown, long_tolerance: Unknown = False) -> Unknown"
  6:13 - info: Type of "jellyfish.damerau_levenshtein_distance" is "(s1: Unknown, s2: Unknown) -> Unknown"
  7:13 - info: Type of "jellyfish.soundex" is "(s: Unknown) -> Unknown"
  8:13 - info: Type of "jellyfish.hamming_distance" is "(s1: Unknown, s2: Unknown) -> Unknown"
  9:13 - info: Type of "jellyfish.nysiis" is "(s: Unknown) -> Unknown"
  10:13 - info: Type of "jellyfish.match_rating_codex" is "(s: Unknown) -> Unknown"
  11:13 - info: Type of "jellyfish.match_rating_comparison" is "(s1: Unknown, s2: Unknown) -> Unknown"
  12:13 - info: Type of "jellyfish.metaphone" is "(s: Unknown) -> Unknown"
  13:13 - info: Type of "jellyfish.porter_stem" is "(s: Unknown) -> Unknown"
0 errors, 0 warnings, 11 infos
Completed in 0.625sec

$ npx pyright@1.1.99 test.py
npx: installed 1 in 2.13s
Loading configuration file at /[...]/test/pyrightconfig.json
Assuming Python version 3.9
Assuming Python platform Darwin
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding .git
stubPath /[...]/test/typings is not a valid directory.
Searching for source files
Found 1 source file
/[...]/test/test.py
  3:13 - info: Type of "jellyfish.levenshtein_distance" is "(s1: Unknown, s2: Unknown) -> int"
  4:13 - info: Type of "jellyfish.jaro_similarity" is "(s1: Unknown, s2: Unknown) -> float"
  5:13 - info: Type of "jellyfish.jaro_winkler_similarity" is "(s1: Unknown, s2: Unknown, long_tolerance: Unknown = False) -> float"
  6:13 - info: Type of "jellyfish.damerau_levenshtein_distance" is "(s1: Unknown, s2: Unknown) -> int"
  7:13 - info: Type of "jellyfish.soundex" is "(s: Unknown) -> str"
  8:13 - info: Type of "jellyfish.hamming_distance" is "(s1: Unknown, s2: Unknown) -> int"
  9:13 - info: Type of "jellyfish.nysiis" is "(s: Unknown) -> str"
  10:13 - info: Type of "jellyfish.match_rating_codex" is "(s: Unknown) -> str"
  11:13 - info: Type of "jellyfish.match_rating_comparison" is "(s1: Unknown, s2: Unknown) -> bool | None"
  12:13 - info: Type of "jellyfish.metaphone" is "(s: Unknown) -> str"
  13:13 - info: Type of "jellyfish.porter_stem" is "(s: Unknown) -> str"
0 errors, 0 warnings, 11 infos
Completed in 0.783sec
```

**Expected behavior**
That Pyright would be able to infer the return type of `jellyfish` functions as before.

**Screenshots or Code**
```python
import jellyfish

reveal_type(jellyfish.levenshtein_distance)
reveal_type(jellyfish.jaro_similarity)
reveal_type(jellyfish.jaro_winkler_similarity)
reveal_type(jellyfish.damerau_levenshtein_distance)
reveal_type(jellyfish.soundex)
reveal_type(jellyfish.hamming_distance)
reveal_type(jellyfish.nysiis)
reveal_type(jellyfish.match_rating_codex)
reveal_type(jellyfish.match_rating_comparison)
reveal_type(jellyfish.metaphone)
reveal_type(jellyfish.porter_stem)

```

**VS Code extension or command-line**
Either, 1.1.100

**Comment by erictraut** [comment]
Thanks for the bug report, and sorry about the regression. Our unit tests and smoke tests don't cover this particular combination (non-py.test package with no stubs and with useLibraryCodeForTypes enabled), and I ended up breaking it with a recent change. I'll do a quick release of 1.1.101 to address it.

**Comment by erictraut** [comment]
This is now fixed in Pyright 1.1.101, which I just published. It will also be included in the next release of Pylance.

**Comment by layday** [comment]
Thanks for the quick fix!
